scripts/sdk-compatibility-generator.ts

```diff
@@ -0,0 +1,70 @@
+import { writeFileSync } from 'fs';
+import { SdkCompatibility } from '../src/datasets/types';
+import { features } from '../src/datasets/constants';
+import { SDK } from '../src/datasets/sdks';
+
+type Input = Pick<SDK, 'name' | 'category'> & {
+  path: string;
+  repoUrl: string;
+  content: string;
+};
+
+export class SdkCompatibilityGenerator {
+  private _matrix: SdkCompatibility[] = [];
+
+  addSdk(sdk: Input) {
+    this._matrix.push({
+      name: sdk.name,
+      path: sdk.path,
+      category: sdk.category,
+      release: this.getReleaseInfo(sdk.repoUrl, sdk.content),
+      spec: this.getSpecInfo(sdk.content),
+      features: this.getFeatureStatus(sdk.path, sdk.content),
+    });
+  }
+
+  generateJson(path: string) {
+    writeFileSync(path, JSON.stringify(this._matrix, null, 2));
+  }
+
+  private getReleaseInfo(url: string, content: string): SdkCompatibility['release'] {
+    const releaseRegex = new RegExp(
+      `<a href="(${url.replaceAll('/', '\\/').replaceAll('.', '\\.')}\\/releases\\/tag\\/.*(v?\\d+\\.\\d+\\.\\d+))">`
+    );
+    const releaseInfo = releaseRegex.exec(content);
+    return {
+      href: releaseInfo[1],
+      version: releaseInfo[2],
+      stable: !releaseInfo[2].startsWith('0.'),
+    };
+  }
+
+  private getSpecInfo(content: string): SdkCompatibility['spec'] {
```

Comment:
It would be nice to put a comment above these explaining exactly what pattern they're matching.